From 5389c0ada76f2a0e30e9d9ed74e1f32518228e7a Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 07 Aug 2018 16:27:49 +0300
Subject: [PATCH] BTL: pcie phy: fix default registers values

Several default registers values of the PCIe PHY were found incorrect.
Adjust those registers early enough to correct the PCIe PHY configuration.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/board/habanalabs/dali/btl/pcie_phy.s b/board/habanalabs/dali/btl/pcie_phy.s
index f500f59..5fc8dd9 100644
--- a/board/habanalabs/dali/btl/pcie_phy.s
+++ b/board/habanalabs/dali/btl/pcie_phy.s
@@ -19,6 +19,28 @@
 cfg 0xCC10840, 0x0
 #endif
 
+/* Adjust the incorrect PHY protocol0 defaults */
+/* writel(0xA067, mmPCIE_PHY_PROTOCOL0_EXT_MPLLA_BANDWIDTH); */
+cfg 0xCC1010C, 0x0000A067
+/* writel(0x19, mmPCIE_PHY_PROTOCOL0_EXT_MPLLA_MULTIPLIER); */
+cfg 0xCC10128, 0x00000019
+/* writel(0xA01E, mmPCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH); */
+cfg 0xCC10140, 0x0000A01E
+/* writel(0x50, mmPCIE_PHY_PROTOCOL0_EXT_MPLLB_MULTIPLIER); */
+cfg 0xCC10158, 0x00000050
+/* writel(0xE, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G1_SLICE0); */
+cfg 0xCC1029C, 0x0000000E
+/* writel(0xC, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G2_SLICE0); */
+cfg 0xCC102A4, 0x0000000C
+/* writel(0xA, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G3_SLICE0); */
+cfg 0xCC102AC, 0x0000000A
+/* writel(0x9, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G4_SLICE0); */
+cfg 0xCC102B4, 0x00000009
+/* writel(0x4, mmPCIE_PHY_PROTOCOL0_EXT_TX_VBOOST_LVL); */
+cfg 0xCC10364, 0x00000004
+/* writel(0x1, mmPCIE_PHY_PHY_EXT_CTRL_SEL); */
+cfg 0xCC10640, 0x00000001
+
 /* Clear phy_init (removes the workaround provided by SNPS) */
 cfg 0xCC07100, 0x00000000
 
